lib/framework/view/footer.dart

```diff
@@ -0,0 +1,233 @@
+import 'package:ensemble/layout/templated.dart';
+import 'package:ensemble/page_model.dart';
+import 'package:ensemble/screen_controller.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble/widget/helpers/controllers.dart';
+import 'package:ensemble/widget/helpers/widgets.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:flutter/material.dart';
+import 'package:ensemble/framework/action.dart' as action;
+import 'package:yaml/yaml.dart';
+import '../../layout/list_view.dart' as ensemblelistview;
+import '../../layout/grid_view.dart' as ensemblegridview;
+import '../../layout/box/box_layout.dart' as ensemblecolumn;
+
+class Footer extends StatefulWidget
+    with
+        UpdatableContainer,
+        Invokable,
+        HasController<FooterController, _FooterState> {
+  Footer({super.key});
+
+  static const String type = "footer";
+
+  @override
+  State<Footer> createState() => _FooterState();
+
+  @override
+  FooterController get controller => _controller;
+
+  final FooterController _controller = FooterController();
+
+  @override
+  Map<String, Function> getters() {
+    return {"scrollBehaviour": () => controller.scrollBehavior};
+  }
+
+  @override
+  Map<String, Function> methods() {
+    return {};
+  }
+
+  @override
+  Map<String, Function> setters() {
+    return {
+      "children": (value) =>
+          _controller.dragableChildrenMap = Utils.getYamlMap(value),
+      "dragOptions": (value) => _controller.dragOptions =
+          DragOptions.fromYaml(Utils.getYamlMap(value) ?? YamlMap(), this)
+    };
+  }
+
+  @override
+  void initChildren({List<WidgetModel>? children, ItemTemplate? itemTemplate}) {
+    _controller.children = children;
+  }
+}
+
+class _FooterState extends WidgetState<Footer>
+    with HasChildren<Footer>, TemplatedWidgetState {
+  DragOptions? _dragOptions;
+  late DraggableScrollableController _dragController;
+  @override
+  void initState() {
+    super.initState();
+    _dragController = widget._controller.dragController;
+    _dragOptions = widget.controller.dragOptions;
+    if (_dragOptions != null) {
+      _dragOptions = widget._controller.dragOptions!;
+      _dragController.addListener(
+        () {
+          if (_dragController.size == _dragOptions?.maxSize &&
+              _dragOptions?.onMaxSize != null) {
+            ScreenController().executeAction(context, _dragOptions!.onMaxSize!);
+          }
+          if (_dragController.size == _dragOptions?.minSize &&
+              _dragOptions?.onMinSize != null) {
+            ScreenController().executeAction(context, _dragOptions!.onMinSize!);
+          }
+        },
+      );
+    }
+  }
+
+  @override
+  void didUpdateWidget(covariant Footer oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    widget._controller.dragController = oldWidget._controller.dragController;
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _dragController.dispose();
```

Comment:
call `super.dispose` is suppose to call in end, after all things of call are dispose